### DISCUSSION TITLE
New schintilla like editor?

### DISCUSSION THREAD

**Discussion by Banaanae** [mmikeww/AHK-v2-script-converter#211] [Ideas]
The current schintilla editor is quite buggy (experimental), it might be an idea to use a different library

As for the library itself I haven't looked into them, might be worth looking into Monaco editor, as it provides a good editor and diffing, but it's written in JS (I have no experience running Js inside ahk)

**Comment by fade2gray** [comment]
Schintilla ⇒ Scintilla 😉

Ever thought about developing an extension for VSCode?

**Reply by Banaanae** [reply]
100% possible, I just don't like the idea of needing to maintain 2 different code bases in 2 different languages that both at the end of the day do the same thing. I use VSCode, but many people use other IDEs ([SciTE4AutoHotkey](https://www.autohotkey.com/scite4ahk/), Sublime text), which may or may not support VSCode extensions.

The other thing is the porting process wouldn't be that easy, while majority of functions can be easily swapped with js counterparts, there will be issues with many of the advanced RegExs, which will require addition changes.

I'd only see this happening once the converter is very close to a one-click solution (which I forgot to mention, but the main reason I'd like something like this is when I'm finding the correct conversion to an issue, and I easily get lost in the sea of text (especially when the edit box is 200px wide D:), and once the converter reaches this point I would be needing to do that as often)

**Reply by CoffeeChaton** [reply]
> 100% possible, I just don't like the idea of needing to maintain 2 different code bases in 2 different languages that both at the end of the day do the same thing. I use VSCode, but many people use other IDEs ([SciTE4AutoHotkey](https://www.autohotkey.com/scite4ahk/), Sublime text), which may or may not support VSCode extensions.
> 
> The other thing is the porting process wouldn't be that easy, while majority of functions can be easily swapped with js counterparts, there will be issues with many of the advanced RegExs, which will require addition changes.

The extension of vscode can just write the LSP client, which only accepts, parses and converts json into a format that vscode can accept. 

It is not necessary or recommended to use js to re-implement this project.

**Comment by mmikeww** [comment]
the mergely diff tool in the legacy version of the converter is written in JS and called by AHK, i forget how i originally did it but should be in there

**Reply by Banaanae** [reply]
I'll look into it, thanks!

**Comment by Banaanae** [comment]
![image](https://github.com/mmikeww/AHK-v2-script-converter/assets/83927639/87042a77-2983-4aff-b853-917684f1a7c4)
Quick poc with monaco editor and this [WebView2 lib](https://github.com/thqby/ahk2_lib/tree/master/WebView2)
